pkg/kubeapply/waiter.go

```diff
@@ -87,32 +93,38 @@ func (w *Waiter) isEmpty() bool {
 func (w *Waiter) Wait(ctx context.Context, deadline time.Time) (bool, error) {
 	start := time.Now()
 	printed := make(map[string]bool)
-	err := w.watcher.WatchQuery(k8s.Query{Kind: "events", Namespace: k8s.NamespaceAll}, func(watcher *k8s.Watcher) error {
-		list, err := watcher.List("events")
+	err := w.watcher.WatchQuery(k8s.Query{Kind: "Events.v1.", Namespace: k8s.NamespaceAll}, func(watcher *k8s.Watcher) error {
+		list, err := watcher.List("Events.v1.")
 		if err != nil {
 			return err
 		}
-		for _, r := range list {
-			if lastStr, ok := r["lastTimestamp"].(string); ok {
-				last, err := time.Parse("2006-01-02T15:04:05Z", lastStr)
-				if err != nil {
-					dlog.Println(ctx, err)
-					continue
-				}
-				if last.Before(start) {
-					continue
-				}
+		for _, _event := range list {
+			var event corev1.Event
+			if err := kates_internal.Convert(_event, &event); err != nil {
```

Comment:
Worth a comment here about what type `_event` is.

Reply:
An `interface{}`!